src/Data/Parity/Properties.agda

```diff
@@ -420,58 +417,137 @@ toSign-injective {p} {q} eq = begin
 toSign-isMagmaMonomorphism = record
   { isMagmaHomomorphism = toSign-isMagmaHomomorphism
   ; injective = toSign-injective
   }
 
-toSign-isMagmaIsomorphism : IsMagmaIsomorphism +-rawMagma Sign.*-rawMagma toSign
+toSign-isMagmaIsomorphism : IsMagmaIsomorphism ℙ.+-rawMagma 𝕊.*-rawMagma toSign
 toSign-isMagmaIsomorphism = record
   { isMagmaMonomorphism = toSign-isMagmaMonomorphism
   ; surjective = toSign-surjective
   }
 
-toSign-isMonoidHomomorphism : IsMonoidHomomorphism +-0-rawMonoid Sign.*-1-rawMonoid toSign
+toSign-isMonoidHomomorphism : IsMonoidHomomorphism ℙ.+-0-rawMonoid 𝕊.*-1-rawMonoid toSign
 toSign-isMonoidHomomorphism = record
   { isMagmaHomomorphism = toSign-isMagmaHomomorphism
   ; ε-homo = refl
   }
 
-toSign-isMonoidMonomorphism : IsMonoidMonomorphism +-0-rawMonoid Sign.*-1-rawMonoid toSign
+toSign-isMonoidMonomorphism : IsMonoidMonomorphism ℙ.+-0-rawMonoid 𝕊.*-1-rawMonoid toSign
 toSign-isMonoidMonomorphism = record
   { isMonoidHomomorphism = toSign-isMonoidHomomorphism
   ; injective = toSign-injective
   }
 
-toSign-isMonoidIsomorphism : IsMonoidIsomorphism +-0-rawMonoid Sign.*-1-rawMonoid toSign
+toSign-isMonoidIsomorphism : IsMonoidIsomorphism ℙ.+-0-rawMonoid 𝕊.*-1-rawMonoid toSign
 toSign-isMonoidIsomorphism = record
   { isMonoidMonomorphism = toSign-isMonoidMonomorphism
   ; surjective = toSign-surjective
   }
 
-toSign-isGroupHomomorphism : IsGroupHomomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupHomomorphism : IsGroupHomomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupHomomorphism = record
   { isMonoidHomomorphism = toSign-isMonoidHomomorphism
   ; ⁻¹-homo = ⁻¹-homo-opposite
   }
 
-toSign-isGroupMonomorphism : IsGroupMonomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupMonomorphism : IsGroupMonomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupMonomorphism = record
   { isGroupHomomorphism = toSign-isGroupHomomorphism
   ; injective = toSign-injective
   }
 
-toSign-isGroupIsomorphism : IsGroupIsomorphism +-0-rawGroup Sign.*-1-rawGroup toSign
+toSign-isGroupIsomorphism : IsGroupIsomorphism ℙ.+-0-rawGroup 𝕊.*-1-rawGroup toSign
 toSign-isGroupIsomorphism = record
   { isGroupMonomorphism = toSign-isGroupMonomorphism
   ; surjective = toSign-surjective
   }
 
+
+------------------------------------------------------------------------
+-- relating Nat and Parity
+
+-- successor and (_⁻¹)
+
+suc-homo-⁻¹ : ∀ n → (parity (suc n)) ⁻¹ ≡ parity n
+suc-homo-⁻¹ zero    = refl
+suc-homo-⁻¹ (suc n) = ⁻¹-inverts (suc-homo-⁻¹ n)
+
+-- parity is a _+_ homomorphism
+
++-homo-+      : ∀ m n → parity (m ℕ.+ n) ≡ parity m ℙ.+ parity n
++-homo-+ zero    n = refl
++-homo-+ (suc m) n = begin
+  parity (suc m ℕ.+ n)        ≡⟨ suc-+-homo-⁻¹ m n ⟩
+  (parity m) ⁻¹ ℙ.+ parity n   ≡⟨ cong (ℙ._+ parity n) (suc-homo-⁻¹ (suc m)) ⟩
+  parity (suc m) ℙ.+ parity n  ∎
+  where
+    open ≡-Reasoning
+    suc-+-homo-⁻¹ : ∀ m n → parity (suc m ℕ.+ n) ≡ (parity m) ⁻¹ ℙ.+ parity n
```

Comment:
I would probably export this lemma as well, instead of burying it here?

Reply:
Here, I disagree, or at least: if I were to do so, it would reinstate the mutual dependency between this lemma (clearly auxiliary to `+-homo-+`) which @MatthewDaggitt objected to above. On balance, I though he was correct. The other lemma, `suc-homo-⁻¹`, on th eother hand, is independently useful, I think.

Reply:
Fair enough.